src/System.CommandLine/Option.cs

```diff
@@ -133,8 +133,11 @@ internal virtual Argument Argument
         /// </value>
         public string ArgumentHelpName
         {
-            get => Argument.Name;
-            set => Argument.Name = value;
+            get => Argument.HelpName ?? Argument.Name;
```

Comment:
I wonder if we should skip the fallback value and move the fallback logic to the `HelpBuilder`.

Reply:
Yeah, that makes sense to me. I was going to make the type here `string?` and not have this default but that breaks this test: https://github.com/dotnet/command-line-api/blob/d6d49b068ffa199ca60a2aa4a2d31e748c4dbe73/src/System.CommandLine.Tests/OptionTests.cs#L239, are you okay with that behavior change?

Reply:
If that's the only test that breaks, that seems fine.

Reply:
Sounds good, pushed that change along with other suggestions.